src/components/file-input/README.md

```diff
@@ -1 +1,99 @@
+    <svg role="img" alt="" xmlns="http://www.w3.org/2000/svg" width="16" height="16" viewBox="0 0 16 16">
+        <path fill="currentColor" d="M10.41,1H3.5A1.3,1.3,0,0,0,2.2,2.3V13.7A1.3,1.3,0,0,0,3.5,15h9a1.3,1.3,0,0,0,1.3-1.3V4.39ZM11.8,5.21V6H9.25V3h.34ZM4.2,13V3h3V6.75A1.25,1.25,0,0,0,8.5,8h3.3v5Z"/>
+    </svg>
+</label>
+```
+
+**`div`**:
+
+```
+<div class="rvt-file__preview" data-upload-preview="fileInputId" id="file-description">
+    No file selected
+</div>
+```
+
+The elements `div[data-upload-wrapper]`, `input[data-upload-input]`, `input[id]`, and `div[data-upload-preview]` should all have the same unique `id`.
+
+The attribute `input[aria-describedby]` should match `div[id]` for assistive technology to properly read the file description.
+
+```
+<div class="rvt-file" data-upload-wrapper="fileInputId">
+    <input type="file" data-upload-input="fileInputId" id="fileInputId" aria-describedby="file-description">
+    <label for="fileInputId" class="rvt-button">
+        <span>Upload a file</span>
+        <svg role="img" alt="" xmlns="http://www.w3.org/2000/svg" width="16" height="16" viewBox="0 0 16 16">
+            <path fill="currentColor" d="M10.41,1H3.5A1.3,1.3,0,0,0,2.2,2.3V13.7A1.3,1.3,0,0,0,3.5,15h9a1.3,1.3,0,0,0,1.3-1.3V4.39ZM11.8,5.21V6H9.25V3h.34ZM4.2,13V3h3V6.75A1.25,1.25,0,0,0,8.5,8h3.3v5Z"/>
+        </svg>
+    </label>
+    <div class="rvt-file__preview" data-upload-preview="fileInputId" id="file-description">
+        No file selected
+    </div>
+</div>
+```
+
+## Initialization
+
+For each file input on the page:
+- Use the `[data-upload-wrapper]` attribute to select the file input.
+- Initialize the file input using the selected file input from the previous step (referenced as `selectedFileInput`):
+    - `const newFileInput = new Rivet.FileInput(selectedFileInput)`
+
+## "Public" methods
+
+If you use the appropriate data attribute/id combination in your markup, the file input will work without the need for any additional JavaScript. But if you need to control the file input programmatically, there are two methods from the Rivet file input's API you can use:
+
+| Method           | Description                                                                            |
+| ---------------- | -------------------------------------------------------------------------------------- |
+| .init()  | Adds the built-in event listeners to the file input |
+| .destroy() | Removes all built-in event listeners from the file input |
+
+## Accessibility Requirements
+
+The Rivet file input is built to follow the WAI-ARIA authoring standards. It is marked up with the appropriate ARIA attributes and uses the JavaScript included in rivet.js to implement the keyboard navigation and focus management required to meet the ARIA Authoring Practices standards. If you need to create the file input functionality in another framework/library like React, Angular, etc., please ensure that it meets the following accessibility requirements.
```

Comment:
I'm not sure this whole part about ARIA authoring practices is needed here. The file input is not any kind of officially documented widget that aria has recommended roles and states for. For example, if you look at the left nav on the ARIA site, there's no link to "file input" or equivalent. Other widgets like Accordion, Modal, Tabs, etc. are more "standar" widgets that ARIA has specs for. Eg. you can [link directly to the recommendations for Accordions](https://w3c.github.io/aria-practices/#accordion)

Our implementation of the file input is mostly custom styling with JavaScript that helps make it more accessible, but the ARIA doesn't necessarily make any recommendations here.

It's still important to have the accessibility requirements section here to call out the focus behavior, keyboard interactions, and any labeling, roles, states, etc. like what you have here for `aria-describedby`.

Reply:
Ah, I understand. I misinterpreted that portion from another component's documentation on rivet.iu.edu as an explanation that we've used `aria` attribute(s) to improve accessibility. This is great to know – I'll adjust it here and keep it in mind for future documentation 👍

Reply:
Yeah, i _think_ the file input in Rivet is the only JS component that is outlier. I believe every other Rivet JS component maps to a specific design pattern/widget in the aria authoring practices docs. 👍